spanner_orm/condition.py

```diff
@@ -173,31 +175,30 @@ def _spanner_type_of_python_object(value: GuessableParamType) -> type_pb2.Type:
     raise TypeError(
         'Cannot infer type of None, because any SQL type can be NULL.')
   simple_type_code = {
-      bool: type_pb2.BOOL,
-      int: type_pb2.INT64,
-      float: type_pb2.FLOAT64,
-      datetime_helpers.DatetimeWithNanoseconds: type_pb2.TIMESTAMP,
-      datetime.datetime: type_pb2.TIMESTAMP,
-      datetime.date: type_pb2.DATE,
-      bytes: type_pb2.BYTES,
-      str: type_pb2.STRING,
-      decimal.Decimal: type_pb2.NUMERIC,
+      bool: spanner_v1.TypeCode.BOOL,
+      int: spanner_v1.TypeCode.INT64,
+      float: spanner_v1.TypeCode.FLOAT64,
+      datetime_helpers.DatetimeWithNanoseconds: spanner_v1.TypeCode.TIMESTAMP,
+      datetime.datetime: spanner_v1.TypeCode.TIMESTAMP,
+      datetime.date: spanner_v1.TypeCode.DATE,
+      bytes: spanner_v1.TypeCode.BYTES,
+      str: spanner_v1.TypeCode.STRING,
+      decimal.Decimal: spanner_v1.TypeCode.NUMERIC,
   }.get(type(value))
   if simple_type_code is not None:
-    return type_pb2.Type(code=simple_type_code)
+    return spanner_v1.Type(code=simple_type_code)
   elif isinstance(value, (list, tuple)):
     element_types = tuple(
         _spanner_type_of_python_object(item)
         for item in value
         if item is not None)
     unique_element_type_count = len({
-        # Protos aren't hashable, so use their serializations.
-        element_type.SerializeToString(deterministic=True)
-        for element_type in element_types
+        # Protos aren't hashable, so serialize them.
+        str(element_type) for element_type in element_types
```

Comment:
Note that `element_type.SerializeToString` does not exist on `spanner_v1.Type`. I'm mildly concerned about the deterministic part and very open to a different method if you know how to find the equivalent of `SerializeToString`.

Reply:
FWIW, the `Type` proto is pretty simple and currently will have a max of two attributes set, both of which are enums, so I don't think there will be issues with determinism here.

Reply:
Alternatively, if you're really concerned about it you could use `element_type._pb.SerializeToString(deterministic=True)` but I don't think it's necessary.

Reply:
I'm not sure `str()` of a proto is meant to be used for anything other than debugging. What do you think about changing this a bit more to not rely on hashing at all? Something like:

```python
if element_types and all(a == b for a, b in zip(element_types, element_types[1:])):
  ...
else:
  ...
```